app/models/envirotech/mappable.rb

```diff
@@ -0,0 +1,72 @@
+module Envirotech
+  module Mappable
+    def self.included(klass)
+      klass.settings = {
+        index: {
+          analysis: {
+            analyzer:
+              { english_analyzer:    {
+                tokenizer: 'standard',
+                filter:    %w(standard asciifolding lowercase snowball) },
+                french_analyzer:     {
+                  tokenizer: 'standard',
+                  filter:    %w(standard asciifolding lowercase french) },
+                chinese_analyzer:    {
+                  tokenizer: 'standard',
+                  filter:    %w(standard  lowercase) },
+                portuguese_analyzer: {
+                  tokenizer: 'standard',
+                  filter:    %w(standard asciifolding lowercase portuguese) },
+                spanish_analyzer:    {
+                  tokenizer: 'standard',
+                  filter:    %w(standard asciifolding lowercase spanish) },
+
+              },
+            filter:   {
+              french:     { type:     'snowball',
+                            language: 'French' },
+              portuguese: { type:     'snowball',
+                            language: 'Portuguese' },
+              spanish:    { type:     'snowball',
+                            language: 'Spanish' },
+            },
+          },
+        },
+      }.freeze
+
+      klass.mappings = {
+        klass.name.typeize => {
+          _timestamp: {
+            enabled: true,
+            store:   true,
+          },
+          properties: {
+            name_chinese:        { type: 'string', analyzer: 'chinese_analyzer' },
+            name_english:        { type: 'string', analyzer: 'english_analyzer' },
+            name_french:         { type: 'string', analyzer: 'french_analyzer' },
+            name_portuguese:     { type: 'string', analyzer: 'portuguese_analyzer' },
+            name_spanish:        { type: 'string', analyzer: 'spanish_analyzer' },
+
+            abstract_chinese:    { type: 'string', analyzer: 'chinese_analyzer' },
+            abstract_english:    { type: 'string', analyzer: 'english_analyzer' },
+            abstract_french:     { type: 'string', analyzer: 'french_analyzer' },
+            abstract_portuguese: { type: 'string', analyzer: 'portuguese_analyzer' },
+            abstract_spanish:    { type: 'string', analyzer: 'spanish_analyzer' },
+
+            source_created_at:   { type: 'date', format: 'YYYY-MM-dd' },
+            source_updated_at:   { type: 'date', format: 'YYYY-MM-dd' },
+            source:              { type: 'string', analyzer: 'keyword' },
+            url:                 { type: 'string' },
+            source_id:           { type: 'string' },
+          },
+        },
+      }.freeze
```

Comment:
At this stage, we don't have any fine-grained requirements about how each endpoint should behave. But it's worth keeping in mind that this isn't quite the same as other consolidated endpoints where several sources are made available via the same endpoint. What we have is several similar sources being made available via several similar endpoints. So it might happen that having the same mappings for each index doesn't work.